From 1453ae0c92927ffdee0fabd67fb94efd1ecd45cf Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:46 -0400
Subject: [usb] usb_wwan: remove redundant urb kill from port remove

Message-id: <1410887463-20674-455-git-send-email-dzickus@redhat.com>
Patchwork-id: 95383
O-Subject: [RHEL7 PATCH 454/591] USB: usb_wwan: remove redundant urb kill from port remove
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 2b4aceabb1bde8fb205f4e947f2db4b1b7a0c2f9
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:23 2014 +0200

    USB: usb_wwan: remove redundant urb kill from port remove

    Remove redundant usb_kill_urb from port remove, which is called
    post-shutdown (close).

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/usb_wwan.c b/drivers/usb/serial/usb_wwan.c
index aa45985..c4a815c 100644
--- a/drivers/usb/serial/usb_wwan.c
+++ b/drivers/usb/serial/usb_wwan.c
@@ -546,20 +546,17 @@ int usb_wwan_port_remove(struct usb_serial_port *port)
  portdata = usb_get_serial_port_data(port);
  usb_set_serial_port_data(port, NULL);
 
- /* Stop reading/writing urbs and free them */
  for (i = 0; i < N_IN_URB; i++) {
-  usb_kill_urb(portdata->in_urbs[i]);
   usb_free_urb(portdata->in_urbs[i]);
   free_page((unsigned long)portdata->in_buffer[i]);
  }
  for (i = 0; i < N_OUT_URB; i++) {
-  usb_kill_urb(portdata->out_urbs[i]);
   usb_free_urb(portdata->out_urbs[i]);
   kfree(portdata->out_buffer[i]);
  }
 
- /* Now free port private data */
  kfree(portdata);
+
  return 0;
 }
 EXPORT_SYMBOL(usb_wwan_port_remove);
-- 
1.7.1